### ISSUE TITLE
Warnings with CUDA in WarpX

### ISSUE THREAD

**Issue by ax3l** [ECP-WarpX/picsar#16]
Hi,

compiling WarpX via `cmake -S . -B build -DWarpX_DIMS=2 -DWarpX_COMPUTE=CUDA -DWarpX_OPENPMD=ON` (latest `development` as of https://github.com/ECP-WarpX/WarpX/commit/f43dac64920270a7035d0e7b9244dcaa89935f0d) on Summit raises the following warnings:
```
/ccs/home/huebl/src/warpx/build/_deps/fetchedpicsar-src/multi_physics/QED/include/picsar_qed/physics/breit_wheeler/breit_wheeler_engine_core.hpp(99): warning #3011-D: a host variable "picsar::multi_physics::phys::heaviside_lorentz_electron_rest_energy [with RealType=RealType]" cannot be directly read in a device function
          detected during instantiation of "__nv_bool picsar::multi_physics::phys::breit_wheeler::evolve_optical_depth(RealType, RealType, RealType, RealType &, const TableType &, RealType) [with RealType=amrex::Real, TableType=BW_dndt_table_view, UnitSystem=picsar::multi_physics::phys::unit_system::SI]" 
/ccs/home/huebl/src/warpx/Source/Particles/ElementaryProcess/QEDInternals/BreitWheelerEngineWrapper.H(157): here

/ccs/home/huebl/src/warpx/build/_deps/fetchedpicsar-src/multi_physics/QED/include/picsar_qed/physics/breit_wheeler/breit_wheeler_engine_core.hpp(100): warning #3011-D: a host variable "picsar::multi_physics::phys::heaviside_lorentz_electron_rest_energy [with RealType=RealType]" cannot be directly read in a device function
          detected during instantiation of "__nv_bool picsar::multi_physics::phys::breit_wheeler::evolve_optical_depth(RealType, RealType, RealType, RealType &, const TableType &, RealType) [with RealType=amrex::Real, TableType=BW_dndt_table_view, UnitSystem=picsar::multi_physics::phys::unit_system::SI]" 
/ccs/home/huebl/src/warpx/Source/Particles/ElementaryProcess/QEDInternals/BreitWheelerEngineWrapper.H(157): here
```

**Comment by lucafedeli88** [comment]
Thanks for posting this issue. I will have a look into that as soon as possible (most likely next Monday)

**Comment by lucafedeli88** [comment]
Uhm, I am a bit confused. 

`heaviside_lorentz_electron_rest_energy` is a constant expression, defined in `multi_physics/QED/include/picsar_qed/physics/unit_conversion.hpp` as follows:

```
    template <typename RealType>
    constexpr RealType heaviside_lorentz_electron_rest_energy = RealType(
        phys::electron_mass<double>*phys::light_speed<double>*phys::light_speed<double>*
        conv<quantity::energy, unit_system::SI, unit_system::heaviside_lorentz, double>::fact());
```
According to [NVIDIA docs](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#const-variables) constant floats or doubles can be directly used in device code...

@ax3l , I remember that we discussed about this in the past and that you proposed me a solution, but I forgot what it was :sweat_smile:  . 

It's weird because in `WarpX` we use host constants in device code all the time...

**Comment by ax3l** [comment]
Oh, I think that's a hickup we saw already for a couple of years with complex cuda code.
The work-around is usually to store the variable in a local `constexpr` or `const` inside the kernel and then use that local variable for compute.

I pushed a fix in #17 :)
